awx/ui/src/screens/Instances/Shared/RemoveInstanceButton.test.js

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import { within, render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { InstanceGroupsAPI } from 'api';
+import RemoveInstanceButton from './RemoveInstanceButton';
+import { I18nProvider } from '@lingui/react';
+import { i18n } from '@lingui/core';
+import { en } from 'make-plural/plurals';
+import english from '../../../../src/locales/en/messages';
+
+jest.mock('api');
+
+const instances = [
+  {
+    id: 1,
+    type: 'instance',
+    url: '/api/v2/instances/1/',
+    capacity_adjustment: '0.40',
+    version: '13.0.0',
+    capacity: 10,
+    consumed_capacity: 0,
+    percent_capacity_remaining: 60.0,
+    jobs_running: 0,
+    jobs_total: 68,
+    cpu: 6,
+    node_type: 'execution',
+    node_state: 'ready',
+    memory: 2087469056,
+    cpu_capacity: 24,
+    mem_capacity: 1,
+    enabled: true,
+    managed_by_policy: true,
+  },
+  {
+    id: 2,
+    type: 'instance',
+    url: '/api/v2/instances/2/',
+    capacity_adjustment: '0.40',
+    version: '13.0.0',
+    capacity: 10,
+    consumed_capacity: 0,
+    percent_capacity_remaining: 60.0,
+    jobs_running: 0,
+    jobs_total: 68,
+    cpu: 6,
+    node_type: 'control',
+    node_state: 'ready',
+    memory: 2087469056,
+    cpu_capacity: 24,
+    mem_capacity: 1,
+    enabled: true,
+    managed_by_policy: false,
+  },
+];
+describe('<RemoveInstanceButtton />', () => {
+  test('Should open modal and deprovision node', async () => {
```

Comment:
This is an excellent test case. I like how you simulated a user interaction with the component being tested and wrote your assertions to check for relevant component state updates and function callbacks.